test/utils/assertions.ts

```diff
@@ -0,0 +1,80 @@
+import { ChannelParams } from 'lumino-js-sdk';
+import { sleep } from '.';
+import { LuminoNode } from '../../src/types/node';
+import { ChannelState } from '../common';
+
+export function given(node: LuminoNode): ExpectationContext {
+    return new ExpectationContext(node);
+}
+
+export class ExpectationContext {
+    constructor(private node: LuminoNode) {}
+
+    expectChannel(channel: ChannelParams): ChannelExpectation {
+        return new ChannelExpectation(this.node, channel);
+    }
+}
+
+export class ChannelExpectation {
+    constructor(private node: LuminoNode, private channel: ChannelParams) {}
+
+    async toHaveBalance(expected: number, timeout = 1000, retries = 5) {
+        const actual = await this.pollChannelUntil(
+            (channel) => expected === channel.balance,
+            timeout,
+            retries
+        );
+        expect(actual.balance).toEqual(expected);
+    }
+
+    async toHaveDeposit(expected: number, timeout = 1000, retries = 5) {
+        const actual = await this.pollChannelUntil(
+            (channel) => expected === channel.total_deposit,
+            timeout,
+            retries
+        );
+        expect(actual.total_deposit).toEqual(expected);
+    }
+
+    async toBeInState(expected: string, timeout = 1000, retries = 5) {
+        const actual = await this.pollChannelUntil(
+            (channel) => expected === channel.state,
+            timeout,
+            retries
+        );
+        expect(actual.state).toEqual(expected);
+    }
+
+    // TODO: Remove these polling, as it's actually hiding a bug
+    private async pollChannelUntil(
+        predicate: (actual: any) => boolean,
+        timeout = 1000,
+        retries = 5
+    ) {
+        let actual = await this.node.client.sdk.getChannel(this.channel);
+        for (let i = 0; !predicate(actual) && i < retries; i++) {
+            await sleep(timeout);
+            actual = await this.node.client.sdk.getChannel(this.channel);
+        }
+        return actual;
+    }
+
+    async toBe(expected: ChannelState) {
+        const actual = await this.node.client.sdk.getChannel(this.channel);
+        if (actual.token_address !== undefined) {
```

Comment:
This can be change to `if (actual.token_address)` since you need to check for other invalid values too. Same for the other if statements.

Reply:
I found that you are using actual instead of expected, you should check only if it's expected right?

Reply:
Something like `if (expected.token_address !== undefined && expected.token_address !== null) // do check` should do the trick for each check